### PR TITLE
remove pnpm rule, fix utils build

### LINKED ISSUE #31576: npx only-allow pnpm prevents npm package install
The `"preinstall": "npx only-allow pnpm"` script in the e2e package.json files prevent the built packages being installed by npm.

### Steps to reproduce

1. In `packages/js/e2e-utils` run ` rm package-lock.json`
1. In `plugins/woocommerce` run
```
pnpm rebuild --filter @woocommerce/e2e-utils
# get the full path to pnpm
which pnpm
```
3. In `packages/js/e2e-utils` run 
```
# use full path to pnpm
 /Users/ronrennick/.nvm/versions/node/v12.22.1/bin/pnpm pack
cp woocommerce-e2e-utils-0.1.6.tgz /path/to/test/project
```
4. Add `"@woocommerce/e2e-utils": "file:../woocommerce-e2e-utils-0.1.6.tgz"` to the project dependencies
5. Run `npm i`
```
$ npm i

> @woocommerce/e2e-utils@0.1.6 preinstall /path/to/project/node_modules/@woocommerce/e2e-utils
> npx only-allow pnpm

npx: installed 20 in 2.868s
╔═════════════════════════════════════════════════════════════╗
║                                                             ║
║   Use "pnpm install" for installation in this project.      ║
║                                                             ║
║   If you don't have pnpm, install it via "npm i -g pnpm".   ║
║   For more details, go to https://pnpm.js.org/              ║
║                                                             ║
```

If the above steps are repeated without the `preinstall` script the package installs correctly. If you re-use the same test project, delete the package lock and node_modules folder between tests.

#31574 removes the script from the `e2e-environment` package.

### PR DESCRIPTION
**Pull request by rrennick**
### Changes proposed in this Pull Request:

This PR removes the `npx only-allow pnpm` from the e2e packages. The rule prevents installing the packages with `npm` in another project.

A second build issue I found where the `??` ternary operator is not built correctly under `pnpm`. Since `jest` uses Babel with local files the issue only surfaces when the built package is used in a separate project. 

A third issue is that the environment package is relying on the `puppeteer` dependency in `@automattic/puppeteer-utils`. As we are deprecating that, we need to update the dependency.

Closes #31576 .

### How to test the changes in this Pull Request:

1. Run E2E tests

### CODE REVIEW

File: packages/js/e2e-environment/package.json [hunk 1]
```diff
@@ -35,7 +35,8 @@
     "jest-each": "25.5.0",
     "jest-puppeteer": "^4.4.0",
     "node-stream-zip": "^1.13.6",
-	"readline-sync": "^1.4.10",
+    "puppeteer": "2.1.1",
```
Comment (MrJnrman) [inline packages/js/e2e-environment/package.json:38]: The addition of the `puppeteer` dependency seems to be causing an issue with the CI. I think committing the `pnpm-lock.yaml` file from running `pnpm i` in the `plugins/woocommerce` folder may do the trick.

**Review by MrJnrman — CHANGES_REQUESTED** [review]
Great work here Ron! I was able to reproduce and fix the bug identified in #31576 using my own project. I also ran all our tests they all passed.

I only have one change request noted below.

File: packages/js/e2e-environment/package.json (hunk 1 above)
Comment (rrennick) [reply packages/js/e2e-environment/package.json:38]: Thanks @MrJnrman, it passed this time.

Comment (MrJnrman) [reply packages/js/e2e-environment/package.json:38]: Thanks for making the change @rrennick

**Review by MrJnrman — APPROVED** [review]
Everything looks good and CI is happy. Great Job!

**Pull request merged**